speechbrain/lobes/models/transformer/Conformer.py

```diff
@@ -99,20 +142,149 @@ def __init__(
             nn.Dropout(dropout),
         )
 
-    def forward(self, x, mask=None):
+    def _do_conv(self, x, inhibit_padding: bool):
```

Comment:
I don't understand what this is about.

Reply:
In regular, non-DynChunkTrain mode, the entire input enters this function, and the convolution is applied with zero-padding on the sides (automatically, through the parameters of the convolution).

In DynChunkTrain mode, a tensor with all stacked chunks enters this function. Conceptually, for each chunk, we want the left "padding" to be the injected left context (of `(chunk_size - 1) // 2` frames) for this chunk, and the right padding to be 0s.

We would have two ways of doing this:
- simply inject the left context, do the convolution, then truncate it out, which means the convolution gets computed for all the frames in the left context even if we're throwing them away immediately after
- or disable the built-in padding from the convolution (and inject zeros for the right padding manually).

The DynChunkTrain code does the latter.

I will add explanations on this as comments.